xlators/mgmt/glusterd/src/glusterd.h

```diff
@@ -764,18 +764,18 @@ typedef ssize_t (*gd_serialize_t)(struct iovec outmsg, void *args);
     } while (0)
 
 #define RCU_READ_LOCK                                                          \
-    pthread_mutex_lock(&(THIS->ctx)->cleanup_lock);                            \
+    pthread_mutex_lock(&(this->ctx)->cleanup_lock);                            \
```

Comment:
It's a bad practice to use variables not passed as an argument to the macro. I strongly suggest to change the macro signature to:

```c
#define RCU_READ_LOCK(_this) ...
```